debian/base/include/dub

```diff
@@ -126,6 +128,51 @@ def wait_for_service(host, port, timeout):
             return False
 
 
+def check_url_ready(url, timeout):
+    """Waits for a URL to be retrievable.
+
+    Useful to wait for a REST API service, for example.
+
+    Args:
+        url: URL to retrieve.  Expected HTTP status code: 2xx.
+        timeout: Time in secs to wait for the URL to be retrievable.
+
+    Returns:
+        False, if the timeout expires and the URL is not retrievable, True otherwise.
+
+    """
+    parsed = urlparse.urlparse(url)
+    if not parsed.netloc:
+        print("URL %s is malformed." % (url,), file=sys.stderr)
+        sys.exit(1)
+    splitted = parsed.netloc.split(":", 1)
+    host = "localhost"
+    if len(splitted) > 0:
+      host = parsed.netloc.split(":", 1)[0]
+    port = 80
+    if len(splitted) > 1:
+      port = parsed.netloc.split(":", 1)[1]
+    else:
+      if parsed.scheme == "https":
+        port = 443
+
+    # Check if you can connect to the endpoint
+    status = wait_for_service(host, port, timeout)
+
+    if status:
+
+        # Check if service is responding as expected
+        r = requests.get(url)
+        if r.status_code // 100 == 2:
```

Comment:
That part of the code is copied from `cub`'s `kr-ready`.  I actually considered to provide an optional parameter, `status_code` with a default of `200`, but eventually decided to keep things simple for the time being, and the idea to accept any `2xx` codes sounded like a good idea (which I assume was the reason for the related function in `cub`).

I'm happy to change this if need be.